lib/guest.ts

```diff
@@ -3,20 +3,66 @@
  * SPDX-License-Identifier: GPL-3.0-or-later
  */
 import { getBuilder } from '@nextcloud/browser-storage'
+import { TypedEventTarget } from 'typescript-event-target'
 
 const browserStorage = getBuilder('public').persist().build()
 
+/**
+ * This event is emitted when the list of registered views is changed
+ */
+interface UpdateGuestDisplayName extends CustomEvent<never> {
+	type: 'updateDisplayName'
+}
+
+class GuestUser extends TypedEventTarget<{ updateDisplayName: UpdateGuestDisplayName }> {
+
+	displayName: string | null
+	readonly uid: string
+	readonly isAdmin: boolean
+
+	constructor() {
+		super()
+		this.displayName = browserStorage.getItem('guestNickname') || ''
+		this.uid = browserStorage.getItem('guestUid') || self.crypto.randomUUID()
+		this.isAdmin = false
+	}
+
+	setDisplayName(displayName: string): void {
+		this.displayName = displayName
+		browserStorage.setItem('guestNickname', displayName)
+		this.dispatchTypedEvent('updateDisplayName', new CustomEvent('updateDisplayName') as UpdateGuestDisplayName)
+	}
+
+}
+
+let currentUser: GuestUser | null | undefined
```

Comment:
```suggestion
let currentUser: GuestUser | undefined
```

When could this be null?